### ISSUE TITLE
Handling errors in fetch

### ISSUE THREAD

**Issue by FarahZaqout** [FACG6/posts-project#23]
https://github.com/FACG6/posts-project/blob/543acb90ac266203e938b072a03ac282f6506b41/public/js/domSignup.js#L176

The ```.catch()``` method in your fetch will not be able to handle a 4xx or 5xx response from the server. 

This is handled in the ```.then()``` method. 
```js
.then((res) => {
   if (res.status === 500) // handle server error here
   if (res.status === 400) // handle bad request error here
})
```
the catch in fetch will only see a network error.

**Comment by NouraldinS** [comment]
@FarahZaqout I knew that I'd be curious to a few things while reading your issues :p, that's why I am reading them. A. you can select multiple lines and reference them in the issue while holding the shift key and selecting the lines from the repo :p
B. I might not disagree with what you're stating here, someone else might :p
https://softwareengineering.stackexchange.com/questions/350498/handling-errors-in-a-client-server-application

**Comment by FarahZaqout** [comment]
@NoureldeanSaed Yeah, I agree with the post. That's why I'm only checking for status code, a message is unnecessary in these sorts of situations and he can end the response with no worries.
